### PR TITLE
Don't register mouse events if pointer events are supported

### PR DESCRIPTION
**Pull request by rachel-fenichel**
See https://github.com/Microsoft/pxt-blockly/pull/96

@microsoftsam

### CODE REVIEW

File: core/blockly.js [hunk 1]
```diff
@@ -450,10 +450,14 @@ Blockly.bindEventWithChecks_ = function(node, name, thisObject, func,
     }
   };
 
-  node.addEventListener(name, wrapFunc, false);
-  var bindData = [[node, name, wrapFunc]];
+  var bindData = [];
+  // Don't register the mouse event if an equivalent pointer event is supported.
+  if (!window.PointerEvent || !(name in Blockly.Touch.TOUCH_MAP)) {
```
Comment (AnmAtAnm) [inline core/blockly.js]: window might not exist in headless mode. Are we certain this function will never be called in headless? I feel more comfortable checking `window` too.

File: core/blockly.js [hunk 2]
```diff
@@ -495,10 +499,13 @@ Blockly.bindEvent_ = function(node, name, thisObject, func) {
     }
   };
 
-  node.addEventListener(name, wrapFunc, false);
-  var bindData = [[node, name, wrapFunc]];
-
-  // Add equivalent touch event.
+  var bindData = [];
+  // Don't register the mouse event if an equivalent pointer event is supported.
+  if (!window.PointerEvent || !(name in Blockly.Touch.TOUCH_MAP)) {
```
Comment (AnmAtAnm) [inline core/blockly.js]: Same concern with `window`

File: core/blockly.js (hunk 1 above)
Comment (rachel-fenichel) [reply core/blockly.js]: Good point.  That also means that there should be a check for window in touch.js: 
```
Blockly.Touch.TOUCH_MAP = {};
if (window.PointerEvent) {
  Blockly.Touch.TOUCH_MAP = {
    'mousedown': ['pointerdown'],
    'mousemove': ['pointermove'],
    'mouseup': ['pointerup', 'pointercancel']
  };
} else if (goog.events.BrowserFeature.TOUCH_ENABLED) {
  Blockly.Touch.TOUCH_MAP = {
    'mousedown': ['touchstart'],
    'mousemove': ['touchmove'],
    'mouseup': ['touchend', 'touchcancel']
  };
}

```

In the particular case of `bindEvent_`, it's called from inject.js with `window` as the `node` argument, so we may have already failed.

It looks like we used to create a fake window object to solve this problem (not the best solution): https://github.com/google/blockly/pull/1566/files  Now that's gone.

**Comment by rachel-fenichel** [comment]
This still doesn't work--now I get zero clicks in firefox and one click in chrome, instead of one click in firefox and two in chrome.  I'm disabling them for this release.

**Comment by rachel-fenichel** [comment]
Merged, because @samelhusseini proved that the problem was with the firefox simulator.

**Pull request merged**